File: frontend/src/models/Transaction.ts
import Posting from './Posting'

export default class Transaction {
    id: string;
    date: string;
    payee: string;
    payeeComment: string;
    comment: string;
    postings: Posting[];
    isComment: boolean;

    constructor(
        id: string,
        date: string,
        payee: string,
        payeeComment: string,
        comment: string,
        postings: Posting[],
        isComment: boolean
    ) {
        this.id = id;
        this.date = date;
        this.payee = payee;
        this.payeeComment = payeeComment;
        this.comment = comment;
        this.postings = postings;
        this.isComment = isComment;
    }
}

File: internal/models/transaction.go
package models

import (
	"github.com/google/uuid"
)

type Transaction struct {
	Id        uuid.UUID `json:"id"`
	Date      string    `json:"date"`
	Payee     string    `json:"payee"`
	PayeeComment string `json:"payee_comment"`
	Comment   string    `json:"comment"` // omitempty
	Postings  []Posting `json:"postings"`
	IsComment bool      `json:"is_comment"`
}

type Transactions []Transaction
